apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Tests/SubscriberClientTest.cs

```diff
@@ -1231,5 +1296,75 @@ public void ExactlyOnce_Extends_PermanentFault()
                 Assert.True(ex is null);
             });
         }
+
+        [Theory, CombinatorialData]
+        public void ExactlyOnce_AckNack_MixedFault([CombinatorialValues(true, false)] bool ackOrNack)
+        {
+            var msgs = new[]
+            {
+                ServerAction.Data(TimeSpan.Zero, new[] { "1" }),
+                ServerAction.Data(TimeSpan.Zero, new[] { "2" }),
+                ServerAction.Data(TimeSpan.Zero, new[] { "3" }),
+                ServerAction.Inf()
+            };
+
+            // Message 1 is success, 2 is temporary failure and 3 is permanent failure.
+            var exception = GetExactlyOnceMixedException();
+            // We have both temporary and permanent failures.
+            // Permanent failure in ack/nack request should be thrown to the client.
+            // Temporary failure in ack/nack request should be retried.
+            // If ackOrNack is true, then it is acknowledge request. Acknowledge RPC will throw the supplied exception.
+            // If ackOrNack is false, then it is nack request. ModifyAcknowledgeDeadline RPC will throw the supplied exception.
+            var ackModifyAckDeadlineAction =
+                ackOrNack ? AckModifyAckDeadlineAction.BadAck(exception, numberOfFailures: 1)
+                : AckModifyAckDeadlineAction.BadModifyAckDeadline(exception, numberOfFailures: 1);
+
+            using var fake = Fake.Create(new[] { msgs }, useMsgAsId: true, ackModifyAckDeadlineAction: ackModifyAckDeadlineAction, isExactlyOnce: true);
+
+            fake.Scheduler.Run(async () =>
+            {
+                var doneTask = fake.Subscriber.StartAsync(async (msg, ct) =>
+                await Task.FromResult(ackOrNack ? SubscriberClient.Reply.Ack : SubscriberClient.Reply.Nack));
+                await fake.TaskHelper.ConfigureAwait(fake.Scheduler.Delay(TimeSpan.FromSeconds(100), CancellationToken.None));
+                Exception ex = await fake.TaskHelper.ConfigureAwaitHideErrors(() => fake.Subscriber.StopAsync(CancellationToken.None));
+                // For acks or nacks, exception should be thrown for only permanent failure.
+                Assert.True(ex is not null);
```

Comment:
I'd just use Assert.NotNull for this. (Or just the IsType check below, which I believe already checks for non-nullity).

Reply:
Done.